sktime/forecasting/model_selection/_tune.py

```diff
@@ -267,7 +267,10 @@ def evaluate_candidates(candidate_params):
         # Select n best forecaster
         self.n_best_forecasters_ = []
         self.n_best_scores_ = []
-        for i in range(self.return_n_best_forecasters):
+        _forecasters_to_return = self.return_n_best_forecasters
```

Comment:
```suggestion
        _forecasters_to_return = min(self.return_n_best_forecasters, len(results.index))
```

Does it fix the issue?

Reply:
Yes this should fix the issue. Should I apply it?

Reply:
Sure, but please check if it breaks black's 88 character limit or not.

Reply:
I believe it is standing at exactly 88